clang/lib/Sema/SemaExpr.cpp

```diff
@@ -8479,6 +8479,18 @@ Sema::CheckAssignmentConstraints(QualType LHSType, ExprResult &RHS,
   LHSType = Context.getCanonicalType(LHSType).getUnqualifiedType();
   RHSType = Context.getCanonicalType(RHSType).getUnqualifiedType();
 
+  // If the LHS is a checked nt array type and the RHS is an unchecked
+  // nt array type with a bounds-safe interface, use the bounds-safe
+  // interface of the RHS to check the types.
+  if ((LHSType->isNtCheckedArrayType() ||
```

Comment:
Is the test isNtCheckedArrayType needed?   I don't think C allows you to make assignments to values with array type.